crates/engine/tree/src/tree/root.rs

```diff
@@ -330,21 +327,13 @@ where
 
         // Dispatch proof gathering for this state update
         scope.spawn(move |_| {
-            let provider = match view.provider_ro() {
-                Ok(provider) => provider,
-                Err(error) => {
-                    error!(target: "engine::root", ?error, "Could not get provider");
-                    return;
-                }
-            };
-
-            // TODO: replace with parallel proof
-            let result = Proof::overlay_multiproof(
-                provider.tx_ref(),
-                // TODO(alexey): this clone can be expensive, we should avoid it
-                input.as_ref().clone(),
-                proof_targets.clone(),
-            );
+            let nodes_sorted = Arc::new(input.nodes.clone().into_sorted());
+            let state_sorted = Arc::new(input.state.clone().into_sorted());
+            let prefix_sets = Arc::new(input.prefix_sets.clone());
```

Comment:
i think we should include these into the `StateRootConfig`, similar to how I did in my playground branch https://github.com/paradigmxyz/reth/blob/209f287d417467317875c09011dfe838f12f154f/crates/engine/tree/src/tree/root.rs#L61-L87

So that we do not clone and sort on every call

Reply:
makes sense, done here https://github.com/paradigmxyz/reth/pull/13260/commits/f99a7e1085b31717e370d5bd7986cbe78e6f3b34